### PR TITLE
(feat): O3-3855- Ability to clear dropdown input

### PR DESCRIPTION
**Pull request by TrevorAntony**
## Summary
It is possible to clear a selected dropdown option in the form using the approach where we have the first selectable option as “empty”.

## Screenshots
Screenshot 2024-09-04 at 11 50 52
Screenshot 2024-09-04 at 11 50 30
Screenshot 2024-09-04 at 11 50 23

## Related Issue
https://openmrs.atlassian.net/browse/O3-3855

### CODE REVIEW

File: src/components/inputs/select/dropdown.component.tsx [hunk 1]
```diff
@@ -46,16 +47,20 @@ const Dropdown: React.FC<FormFieldInputProps> = ({ field, value, errors, warning
...
             id={field.id}
             titleText={<FieldLabel field={field} />}
             label={t('chooseAnOption', 'Choose an option')}
-            items={field.questionOptions.answers
-              .filter((answer) => !answer.isHidden)
-              .map((item) => item.value || item.concept)}
-            itemToString={itemToString}
-            selectedItem={value}
+            items={[
+              { value: '', label: t('selectAnOption', 'Select an option') },
```
Comment (samuelmale) [inline src/components/inputs/select/dropdown.component.tsx:57]: A better way of doing this is augmenting the question answers from [the schema transformer](https://github.com/openmrs/openmrs-esm-form-engine-lib/blob/main/src/transformers/default-schema-transformer.ts#L137-L166).

**Comment by samuelmale** [comment]
@TrevorAntony I'm closing this PR in favour of https://github.com/openmrs/openmrs-esm-form-engine-lib/pull/375

**Comment by TrevorAntony** [comment]
Well noted

**Comment by samuelmale** [comment]
@TrevorAntony for any new upcoming PR(s) from you, if you're working with a fork, make sure you don't develop off the main branch.

**Comment by TrevorAntony** [comment]
Ok will do.

**Pull request closed without merging**